### PR TITLE
Card color picker

### PR DESCRIPTION
**Pull request by tarehart**
Custom card colors, persisted in local storage. Helps with color blind accessibility. Light theme and dark theme are stored separately.

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -27,6 +27,20 @@ function App() {
...
   const [user, setUser] = useState(null);
   const [themeType, setThemeType] = useStorage("theme", "light");
+  const [customLightTheme, setCustomLightTheme] = useState(lightTheme);
+  const [customDarkTheme, setCustomDarkTheme] = useState(darkTheme);
+  const [customColors] = useStorage("customColors", null);
+
+  useEffect(() => {
+    if (customColors) {
+      if (customColors.light) {
+        setCustomLightTheme(createMuiTheme({...lightTheme, setCard: {...lightTheme.setCard, ...customColors.light}}));
```
Comment (ekzhang) [inline src/App.js:37]: I believe the `createMuiTheme` call is redundant here. Doesn't this function just merge in missing properties in the theme?

File: src/components/ColorChoiceDialog.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  const [red, setRed] = useState(theme.setCard.red);
+  const [green, setGreen] = useState(theme.setCard.green);
+  const [purple, setPurple] = useState(theme.setCard.purple);
+
+  function handleClose() {
+    onClose(null);
+  }
+
+  function handleSubmit() {
+    onClose({red: red, green: green, purple: purple});
```
Comment (ekzhang) [inline src/components/ColorChoiceDialog.js:25]: Example of code that is formatted differently than the rest of the codebase, where Prettier is used.

File: src/App.js [hunk 3]
```diff
@@ -27,6 +27,20 @@ function App() {
   const [uid, setUid] = useState(null);
   const [user, setUser] = useState(null);
   const [themeType, setThemeType] = useStorage("theme", "light");
+  const [customLightTheme, setCustomLightTheme] = useState(lightTheme);
+  const [customDarkTheme, setCustomDarkTheme] = useState(darkTheme);
+  const [customColors] = useStorage("customColors", null);
```
Comment (ekzhang) [inline src/App.js:32]: Why are you defining customColors in both the Navbar and the App state? Isn't this a bit broken? I would imagine that you could just have a handler to change customColors in the App state, and the `useEffect` below this line would update the theme automatically whenever the colors change.

File: src/hooks/useStorage.js [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
+  } catch (e) {
+      return possibleJson;
+  }
+}
+
 function useStorage(key, initialValue) {
   const [value, setValue] = useState(() => {
     const value = window.localStorage.getItem(key);
-    return value === null ? initialValue : value;
+    return value === null ? initialValue : objectOrString(value);
```
Comment (ekzhang) [inline src/hooks/useStorage.js:17]: This try-catch seems implicit and a source for bugs. I would prefer if you called `JSON.stringify()` and `JSON.parse()` directly, outside of this hook, if you need to serialize JSON.

File: src/App.js (hunk 1 above)
Comment (tarehart) [reply src/App.js:37]: You're right, I had assumed createMuiTheme had some necessary side effects, but it seems not. Will simplify.

**Comment by tarehart** [comment]
Good comments! I think I've addressed everything.

**Comment by tarehart** [comment]
Glad to contribute to a cool project, thanks for doing the PR!

**Pull request merged**